controllers/windowsmachine_controller.go

```diff
@@ -106,9 +106,10 @@ func (r *WindowsMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		UpdateFunc: func(e event.UpdateEvent) bool {
 			return r.isValidMachine(e.ObjectNew) && isWindowsMachine(e.ObjectNew.GetLabels())
 		},
-		// ignore delete event for all Machines as WMCO does not react to node getting deleted
+		// process delete event
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			return false
+			// for Windows machines only
+			return isWindowsMachine(e.Object.GetLabels())
```

Comment:
Is there a reason not to call `isValidMachine()`?

Reply:
That's a good question; will defer to @sebsoto  for more context. IMO, `isValidMachine()` is too specific as it  consider the existence of the `machine` along with elements like `machine.Status.Phase` and `machine.Status.Addresses` that may no longer be populated when the event arrive, causing the `Delete` event to be missed.

Reply:
If this is an already existing machine that is being deleted, I would think all the fields should already be populated when the event arrives. WDYT?

Reply:
Not quite following your comment. WMCO should only react to Delete events for machines that include the `Windows` OS label.

Reply:
You still are not calling out why r.isValidMachine() is not called as we discussed during the meeting.